app/main/posts/views/filters/active-search-filters.directive.js

```diff
@@ -31,12 +35,70 @@ function ActiveSearchFilters($translate, $filter, PostFilters, _, FilterTransfor
+            var isModeSavedSearch = PostFilters.getMode() === 'savedsearch';
+            // if it's a different saved search, set it in the scope.
+            // else if there is no saved search but we did find that the mode is 'savedsearch' , get it and set in scope
+            if (isModeSavedSearch || savedSearchChanged) {
+                $scope.savedSearch = savedSearchEntity;
+            } else if (!isModeSavedSearch) {
+                $scope.savedSearch = null;
+            }
+
+            /**
+             * This handles the requirement to have saved search filters displayed in a different way
+             * from the rest of the filters.
+             *  - If there is no saved_search in the filters, continue with the normal filters (else)
+             *  - If there is a saved search:
+             *  - - set it in the scope : this is what we will use to display the saved search and its filters in the UI
+             *  - - get clean version (no defaults) of the saved search filters
+             *  - - get a clean activeFilters array that does not include the saved search filters. $scope.activeFilters: this
+             *  is the array we will be using to show the "extra" filters the user can set AFTER they selected  a saved search
+             *  (keep in mind that when a saved search is selected all filters are erased in favor of the saved search)
+             *  - - - Diffing rules: value from activeFilters takes priority over value from search. this is because
+             *  the user will always have selected the value after they select a saved search,meaning they want to change it.
+             *  - - - Diffing rules: if the value is equal, just ignore it/send empty value.
+             *  - - - Diffing rules: return the _difference if the value is not equal, because we will want to show for instance:
+             *  saved search: tag id 1 + filters tag id 2 (so it's not just ignoring the arrays)
+             */
+            activeFilters = _.mapObject(activeFilters, makeArray);
+            if ($scope.savedSearch) {
+                /**
+                 * to handle removal correctly we need to make sure we take currentFilters (which has up to date info) into account,
+                 * because that is where our savedsearch filters will stop being represented when we remove them.
+                 * If there's a key in our current filters that is in the saved search but is not in the active filters at this point,
+                 * it is because it was removed (since before saved search gets assigned, they are all assigned to the filters)
+                 * that means we have to remove it from the saved search.
+                 **/
+                $scope.savedSearch.filter = cleanDeprecatedValuesFromSavedSearch(activeFilters, _.mapObject(PostFilters.getCleanActiveFilters($scope.savedSearch.filter), makeArray));
+                $scope.activeFilters = _.mapObject(activeFilters, function (value, key) {
```

Comment:
Just trying to figure out if I followed whats going on here. Would it be accurate to describe this as splitting the filters? ie. splitting into "saved search filters" and "other filters" ...
Also cleaning out 'removed but not yet saved' filters from the savedsearch... but I have no idea what to call that.

Reply:
right, yea I think that's accurate. It's clearing out filters that don't have to be shown in the "normal" active filters and then also clearing out filters that were removed from the saved search filters. Does that make sense?